### ISSUE TITLE
[BUG]: Valid formats in Calibre rejected by Readarr (KEPUB)

### ISSUE THREAD

**Issue by jeff47** [Readarr/Readarr#1243]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Kepub is a valid Calibre format (via a plugin) but Readarr will not allow it to be selected as the "convert to" format.

### Expected Behavior

Readarr should allow all valid Calibre formats.

### Environment

```markdown
- OS: linux
- Readarr:0.1.0.971
- Docker Install: yes
- Using Reverse Proxy: yes
- Browser: brave
```

### What branch are you running?

Develop

### Anything else?

Nothing else.

[AB#1610](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1610)

**Comment by austinwbest** [comment]
This is not an app that abides by what another app does. There is no plans to support every format calibre does bud.

**Comment by jeff47** [comment]
I appreciate the work on a great app, and I'm disappointed by this flippant response.  Is it because I incorrectly classified it as a bug rather than an enhancement request?

Readarr doesn't do the conversion; Calibre handles them, correct?  Assuming my understanding is correct (and it may not be), allowing additional formats is a matter of letting Readarr know that those formats are valid options and Calibre can handle them.  As it current stands, Readarr is rejecting valid conversion targets and unnecessarily limiting functionality.

Again, perhaps I'm overlooking something but would the change require more than an update to the dialog validation routine?

Is it as simple as adding a format to CalibreFormat in CalibreConversionOptions.cs?

**Comment by ta264** [comment]
AFAIK calibre doesn't have any way to list via the api what plugins are installed and what formats they might support.  Unless I'm wrong about that, or it changes, Readarr won't support any plugin related functionality.

**Comment by jeff47** [comment]
Right, I don't see how Readarr could know about all the possible formats Calibre supports.  But does it have to?  If Readarr can't tell what's supported, then can it be more relaxed about rejecting things?  It's not really a matter of supporting plugin functionality, as opposed to just not rejecting it.

It just doesn't seem to make sense to limit functionality.  Sure, a user might put in an invalid format.  That's true for a lot of configurations options, and it's not always Readarr's responsibility to fix or warn about all possible errors.

**Comment by austinwbest** [comment]
> I'm disappointed by this flippant response. 

I'm not real sure what you want/expect as I simply answered you. You stated you expect readarr to do something and i stated that wasn't the intention or purpose of readarr. Confused how that is rude but we can go with it i guess.

> Is it because I incorrectly classified it as a bug rather than an enhancement request?

Not at all, that was just to classify it properly for searching/sorting

> Right, I don't see how Readarr could know about all the possible formats Calibre supports. But does it have to? 

Based on your OP, yes.. It would. "Readarr should allow all valid Calibre formats."

With that said, I did not close the request and leave it at that response. Simply because it wont support **all** formats as you requested doesn't mean it cant support more. I left it open for ta to chime in as he is far more familiar with the ease of adding formats and which ones are a for sure no-go vs which would be a maybe.

**Comment by jeff47** [comment]
Just to close the circle on this:

> > Right, I don't see how Readarr could know about all the possible formats Calibre supports. But does it have to?

> Based on your OP, yes.. It would. "Readarr should allow all valid Calibre formats."

This depends on the approach.  Readarr doesn't have to know about all valid Calibre formats in order to allow them.  Right now, Readarr rejects what it considers to be invalid formats - but it doesn't have a way of knowing what a valid format is.  Therefore, it may reject a valid format.  

A possible solution would be for Readarr to take a more "agnostic" approach and rely on the user to know what their installation of Calibre is capable of.  Then it would not need to know all the valid formats.  Of course, there is the possibility of a user putting in an invalid format.  A warning could be issued to prompt the user to double check the entry, perhaps.

Locally, this patch was all that was necessary to address my particular issue, but I don't think it represents the best approach because it still requires Readarr to "know" all the valid formats which will likely lead to another user wanting to know why format X is being rejected when their Calibre installation supports it.

```
--- ./Readarr/src/NzbDrone.Core/Books/Calibre/Resources/CalibreConversionOptions.cs     2021-09-28 21:35:02.176622815 -0400
+++ CalibreConversionOptions.cs 2021-09-28 21:36:30.006500187 -0400
@@ -5,6 +5,7 @@
     public enum CalibreFormat
     {
         None,
+        KEPUB,
         EPUB,
         AZW3,
         MOBI,
```

**Comment by bakerboy448** [comment]
Calibre supports kepub natively https://calibre-ebook.com/new-in/seventeen

Thus prior plugin comment is no longer a concern

**Comment by binarymelon** [comment]
> Calibre supports kepub natively https://calibre-ebook.com/new-in/seventeen
> 
> Thus prior plugin comment is no longer a concern

What if they don't have the latest version of Calibre? 😜
